### ISSUE TITLE
Binary image classification using ensemble voting classifier

### ISSUE THREAD

**Issue by MLlearner2020** [hema2107/EnsembleModel#1]
Hello Friends i am trying to create Ensemble Voting classifier for Binary image classification problem. I have created 4 models and they are working good. I want to perform Ensemble techniques like Voting classifier for this Binary image classification problem (Benign and Malignant types). All the code is working fine but not able to fit the model, getting error in that part.

I am in the learning phase so it would be helpful if you can provide solution to this problem

pip install opencv-python
pip install imutils
pip install tensorflow==2.7.0.

import warnings
warnings.filterwarnings('ignore')
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import os
import math
import shutil
import glob
import cv2
import imutils
import seaborn as sns
from sklearn.utils import shuffle
import tensorflow as tf
from tensorflow.keras.preprocessing.image import load_img
from keras.preprocessing.image import ImageDataGenerator
from keras.layers import Conv2D, MaxPool2D, Dropout, Flatten, Dense, BatchNormalization, GlobalAveragePooling2D
from keras.models import Model, Sequential
import keras

import os
import zipfile

from google.colab import drive
drive.mount('/content/gdrive')

#Count images in the two folders: Benign Nodule Case (Non cancerous) and Malignant Nodule Case (Cancerous)

data_dir = '/content/gdrive/MyDrive/tmp/003 Stage 3_80_15_05 Balanced mask image only/Train'

dict_img = {}
for dir in os.listdir(data_dir):
dict_img[dir] = len(os.listdir(os.path.join(data_dir, dir)))

dict_img

height, width = 224, 224
batch_size=32
def image_generator(height,width):
datagen = ImageDataGenerator(
rescale=1./255.,
validation_split=0.2,
shear_range = 0.2,
zoom_range = 0.2,
horizontal_flip = True,
)
train_ds = datagen.flow_from_directory(
data_dir,
batch_size=batch_size,
subset="training",
color_mode = 'rgb',
shuffle=True,
class_mode='binary',
target_size=(224, 224),
classes={'benign': 0., 'malignant': 1.}
)
val_ds = datagen.flow_from_directory(
data_dir,
subset="validation",
#seed=123,
color_mode = 'rgb',
class_mode='binary',
target_size=(224, 224),
batch_size=batch_size,
classes={'benign': 0., 'malignant': 1.}
)
return train_ds, val_ds

train_ds, val_ds = image_generator(height, width)

total_image = np.concatenate([train_ds.labels,val_ds.labels])
print('\n\n',{'Benign cases':len(np.where(total_image==0)[0]),
'Malignant cases':len(np.where(total_image==1)[0])})

from keras.models import Model, load_model
from keras.layers import Input, Average

Load the trained model
model_1 = load_model('/content/gdrive/MyDrive/Colab Notebooks/model/101_model_ResNet50.h5')
model_2 = load_model('/content/gdrive/My Drive/Colab Notebooks/model/102_model_vgg19.h5')
model_3 = load_model('/content/gdrive/My Drive/Colab Notebooks/model/103_model_inception_v3.h5')
model_4 = load_model('/content/gdrive/My Drive/Colab Notebooks/model/105_CNN.h5')

models=[model_1,model_2,model_3,model_4]
model_input=Input(shape=(224,224,3))
model_outputs=[model(model_input) for model in models]

from sklearn.ensemble import VotingClassifier

Create the voting classifier
ensemble_model = VotingClassifier(estimators=[('resnet', model_1), ('vgg', model_2), ('inception', model_3),('CNN', model_4)], voting='soft')

ensemble_model.fit(train_ds,train_ds.labels)
